build.rs

```diff
@@ -156,7 +155,13 @@ fn main() {
         let compiler = cc::Build::new().try_get_compiler().expect(
             "a C compiler is required to compile libbpf-sys using the vendored copy of libbpf",
         );
-        let cflags = compiler.cflags_env();
+        let mut cflags = compiler.cflags_env();
+        println!("cargo:rerun-if-env-changed=LIBBPF_SYS_EXTRA_CFLAGS");
+        let extra_cflags = env::var_os("LIBBPF_SYS_EXTRA_CFLAGS").unwrap_or_default();
+        if !extra_cflags.is_empty() {
+            cflags.push(" ");
+            cflags.push(extra_cflags);
+        }
```

Comment:
More idiomatic way, in my opinion, would be:
```suggestion
        if let Some(extra_cflags) = env::var_os("LIBBPF_SYS_EXTRA_CFLAGS") {
            cflags.push(" ");
            cflags.push(extra_cflags);
        }
```
(untested)

Yes, that is not *exactly* the same semantics: it's different when `LIBBPF_SYS_EXTRA_CFLAGS` is set to empty string, but that should be fine, I'd think.